src/lgar.cxx

```diff
@@ -2430,10 +2655,72 @@ extern double lgar_theta_mass_balance(int layer_num, int soil_num, double psi_cm
     if (psi_cm_loc <= 0 && psi_cm_loc_prev < 1E-50) break;
 
     delta_mass_prev = delta_mass;
+
+
+    //so the above code that avoids infinite loops by breaking in certain cases works, but a fringe case must be addressed. 
+    //When theta is close to theta_r, the loop "while (delta_mass > tolerance)" will attempt to increase psi until mass balance closure occurs, but if the 
+    //AET value is large enough to make theta less than theta_r (or if dzdt was large enough to make the WF move such that the updated theta would be less than theta_r),
+    //the loop would never be able to achieve mass balance closure, and it will break when the one of above conditions is met.
+    //So, the next few lines beginning with "if (delta_mass > tolerance)" addresses the rare case where water would not be able to be extracted without theta becoming less than theta_r anyway
+  }
+
+  if (delta_mass > tolerance){
+    *AET_demand_cm = *AET_demand_cm - fabs(delta_mass - tolerance);
+  }
+
+  if (theta<soil_properties[soil_num].theta_r){
+    printf("moisture less than residual water content \n");
+    printf("soil_properties[soil_num].theta_r: %lf \n", soil_properties[soil_num].theta_r);
+    printf("theta: %lf \n", theta);
+    abort();
   }
   
   return theta;
 
 }
 
+//while this function may or may not be used in LASAM, it will be used in LGARTO
+extern int lgar_correction_type(int num_layers, double* cum_layer_thickness_cm, struct wetting_front* head){
```

Comment:
this function is not used, so we better remove it and save it somewhere else for future development if needed.

Reply:
Agreed